### PR TITLE
Generate trampolines directly inside functions

### LINKED ISSUE #664: Move trampoline functions generation into the signal/whatever using it

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #664.

You can check the generated code [here](https://github.com/gtk-rs/gtk/pull/824).

cc @EPashkin @sdroege

### CODE REVIEW

**Comment by sdroege** [comment]
Generated code looks good, thanks :)

File: src/codegen/trampoline.rs [hunk 1]
```diff
@@ -24,9 +24,8 @@ pub fn generate(
     env: &Env,
     analysis: &Trampoline,
     in_trait: bool,
-    object_name: &str,
+    prepend: &str,
```
Comment (EPashkin) [inline src/codegen/trampoline.rs]: IMHO here better be `indent: usize` as it public function and `prepend` can stay in inner privates.

Comment (EPashkin) [reply src/codegen/trampoline.rs]: I don't insist

**Comment by EPashkin** [comment]
Only small nit and `cargo fmt` run.

File: src/codegen/trampoline.rs (hunk 1 above)
Comment (EPashkin) [reply src/codegen/trampoline.rs]: Hm, all `version_condition` in this file need indent

Comment (GuillaumeGomez) [reply src/codegen/trampoline.rs]: Ok, I'll udpate it!

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks

**Comment by GuillaumeGomez** [comment]
Time for generations!

**Comment by GuillaumeGomez** [comment]
All PRs are now open!

**Comment by EPashkin** [comment]
👍 for all linked PRs

**Pull request merged**